src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -170,25 +169,23 @@ static ucs_status_t uct_ib_mlx5_devx_reg_ksm_data_addr(
     return status;
 }
 
-static ucs_status_t
-uct_ib_mlx5_devx_reg_ksm_data_contig(uct_ib_mlx5_md_t *md,
-                                     uct_ib_mlx5_devx_mr_t *mr, void *address,
-                                     void *iova, int atomic, const char *reason,
-                                     struct mlx5dv_devx_obj **mr_p,
-                                     uint32_t *mkey)
+static ucs_status_t uct_ib_mlx5_devx_reg_ksm_data_contig(
+        uct_ib_mlx5_md_t *md, uct_ib_mlx5_devx_mr_t *mr, void *address,
+        uint64_t iova, int atomic, const char *reason,
+        struct mlx5dv_devx_obj **mr_p, uint32_t *mkey)
 {
     size_t mr_length = mr->super.ib->length;
-    uintptr_t ksm_address;
-    uintptr_t ksm_iova;
+    uint64_t ksm_address;
+    uint64_t ksm_iova;
     size_t ksm_length;
     int list_size;
 
     /* FW requires indirect atomic MR address and length to be aligned
      * to max supported atomic argument size */
-    ksm_address = ucs_align_down_pow2((uintptr_t)address,
-                                      UCT_IB_MD_MAX_MR_SIZE);
-    ksm_iova    = (uintptr_t)UCS_PTR_BYTE_OFFSET(iova, address - ksm_address);
-    ksm_length  = mr_length + iova - (void*)ksm_iova;
+    ksm_address = ucs_align_down_pow2((uint64_t)address, UCT_IB_MD_MAX_MR_SIZE);
+    ksm_iova    = (uint64_t)UCS_PTR_BYTE_OFFSET(iova,
```

Comment:
IMO no need to use UCS_PTR_BYTE_OFFSET here since all terms all uint64_t (after casting address to uint64_t)